pygame removal + driver/library setup automation

jfeldman-dp opened the pull request:
Created a prerequisite script that can be run by users new to patspeak in order to install necessary drivers and python libraries.  

Updated the readme to include brief setup and test execution instructions

Remove pygame and screen.py references within the pat.py file.

jfeldman-dp committed:
Removed all pat.py references to pygame (and screens)

Created a prerequisite script that can be run by users new to patspeak in order to install necessary drivers and python libraries.

Updated the readme to include a setup and execution guide

daurschel reviewed (approved):
Running `prerequisite_setup.py` I do receive an error message.

```
Downloading Kvaser driver from https://www.kvaser.com/download/?utm_source=software&utm_ean=7330130980150&utm_status=latest...
Downloaded kvaser_driver.exe successfully.
Running the installer for kvaser_driver.exe with admin privileges...
Kvaser driver installed successfully.
An unexpected error occurred: [WinError 32] The process cannot access the file because it is being used by another process: 'kvaser_driver.exe'
```

The script still finishes without issue.

daurschel commented on prerequisite_setup.py:
```diff
@@ -0,0 +1,52 @@
+        subprocess.check_call(["powershell", "-Command", f"Start-Process '{file_name}' -Verb runAs"])
+        print("Kvaser driver installed successfully.")
+        
+        # Clean up the installer file
+        os.remove(file_name)
+        print(f"Removed installer file {file_name}.")
```
This file cleanup code does not accomplish anything. It attempts to delete the file immediately after opening it, and fails. I'd recommend a cleaner solution, but for now the .exe doesn't get removed regardless. This code could be deleted.

daurschel commented:
Pardon my nitpicking, it's only a suggestion, and something we can address down the road.

I would even recommend not automating the setup process, and simply adding setup instructions instead. It would be nearly equivalent effort from a user, less "future issue" prone (obfuscation of links, libraries, and files), better for expandability (new libraries in the future).
.
.
.
>>>ADD `requirements.txt` FILE TO REPO

``` bash
python-can==4.5.0
keyboard==0.13.5
canlib==1.29.758
```
.
.
.
>>>ADD SETUP INSTRUCTIONS TO README
## Setup Instructions

The following items are needed to run PATSpeak.
Ensure you have Python 3.x and `pip` installed

### 1. Install Python Libraries

Install all required python libraries from the requirements file:

```bash
pip install -r requirements.txt
```

### 2. Install Kvaser Driver

Go to the link below and install the latest Kvaser Driver:

[https://kvaser.com/single-download/?download_id=47105](https://kvaser.com/single-download/?download_id=47105)

jfeldman-dp committed:
Found another pygame reference through calling events.py.  Commented out the call to events.

Added a "wait" command on the installer.  Now the cleanup command can happen successfully.

jfeldman-dp committed:
Merge branch 'main' into Cleanup-and-PreRequisite-Creation

jfeldman-dp commented:
@daurschel - Appreciate the detailed review. I pushed another commit that fixes the kvaser cleanup error by adding a wait. 

With respect to the recommendation that requirements are written as instructions, that's valid.  The level of maintenance of setup instructions is expected to be low or zero moving forward.  This script took less than 30 min to build and verify,   I would expect it to be pretty simple to keep updated, but will be open to obsoleting it if it proves tedious in the future.

jfeldman-dp commented:
I dropped the ball. thanks for the reminder @daurschel :)

jaronmcdan reviewed (commented):
looks all good
